diff --git a/lib/advanced_roadmap/projects_helper_patch.rb b/lib/advanced_roadmap/projects_helper_patch.rb
index 1bdb00a..33ea7f6 100644
--- a/lib/advanced_roadmap/projects_helper_patch.rb
+++ b/lib/advanced_roadmap/projects_helper_patch.rb
@@ -13,19 +13,32 @@ module AdvancedRoadmap
   module ProjectsHelperPatch
     def self.included(base)
       base.class_eval do
-        def project_settings_tabs_with_more_tabs
-          tabs = project_settings_tabs_without_more_tabs
-          index = tabs.index({:name => 'versions', :action => :manage_versions,
-                              :partial => 'projects/settings/versions', :label => :label_version_plural})
-          if index
-            tabs.insert(index, {:name => 'milestones', :action => :manage_milestones,
-                                :partial => 'projects/settings/milestones',
-                                :label => :label_milestone_plural})
-            tabs.select {|tab| User.current.allowed_to?(tab[:action], @project)}
+
+        def project_settings_tabs_with_advanced_roadmap
+          tabs = project_settings_tabs_without_advanced_roadmap
+          if User.current.allowed_to?(:manage_milestones, @project)
+            options = {:name => 'versions', :action => :manage_versions,
+                       :partial => 'projects/settings/versions',
+                       :label => :label_version_plural}
+            index = tabs.index(options)
+            unless index # Needed for Redmine v3.4.x
+              options[:url] = {:tab => 'versions',
+                               :version_status => params[:version_status],
+                               :version_name => params[:version_name]}
+              index = tabs.index(options)
+            end
+            if index
+              tabs.insert(index + 1,
+                          {:name => 'milestones', :action => :manage_milestones,
+                           :partial => 'projects/settings/milestones',
+                           :label => :label_milestone_plural})
+              tabs.select {|tab| User.current.allowed_to?(tab[:action], @project)}
+            end
           end
           return(tabs)
         end
-        alias_method_chain :project_settings_tabs, :more_tabs
+        alias_method_chain :project_settings_tabs, :advanced_roadmap
+
       end
     end
   end
